_posts/2015-06-11-docker.md

```diff
@@ -0,0 +1,115 @@
+---
+layout: post
+title: "Improving the development workflow with Docker"
+date: "2015-06-11 10:51:25"
+icon: ship
+tags: docker
+categories: tech-stories
+authors: ["Christian Hoffmeister"]
+---
+
+At the moment [Docker][docker] is one of the hottest topics for devops as well as for developers.
+In a nutshell, one can describe Docker as [Linux containers][lxc] with some extra convenience.
+Docker can be classified somewhere in between fully fledged virtual machines and a simple filesystem virtualisation with chroot.
+Since it is not yet clear, if Docker will find it's way into our production deployment, this post only concentrates on how our development workflow benefits from Docker.
+
+# The challenge
+
+A simplified version of our current project is the following: We are developing a [NodeJS][nodejs]-based application that uses a [Java][java]-based backend for storing data which itself uses a [MySQL][mysql] database.
+Pretty standard stuff.
+So far we deployed the backend service via continuous deployment to a central staging machine and this address was hard coded into our frontend project.
+It worked but still could be improved: Backend changes that were not backwards compatible necessarily blocked our frontend developers and forced them, to adapt to these changes before continuing other work.
+
+# The solution
+
+To improve this, we decided to use Docker to make starting everything on our local machine easy.
+
+## Dockerising
+
+First step towards this goal: _dockerise_ our services.
+We had to provide some kind of instructions, that turns a given base image into an image that contains everything needed, to run the given service.
+With regards to Docker, this is a file called `Dockerfile` that we added to all of our individual projects.
+For our frontend project, it looks somewhat like the following example:
+
+{% highlight text %}
+FROM node:0.12.3
+
+ADD . /build
+WORKDIR /build
+
+RUN npm install
+RUN npm run build
+
+CMD ["node", "index.js"]
+{% endhighlight %}
+
+The four main steps here are:
+
+1. Start from the `node:0.12.3` base image from the [offical Docker Hub][docker-hub].
+2. Add the current directory (which is the project folder containing our frontend project sources) to the container.
```

Comment:
I do not believe source (as in source code) has a plural, although I am not sure. I would say:
"(which is the project folder containing our frontend project source code)"